src/main/scala/com/advancedtelematic/campaigner/data/DataType.scala

```diff
@@ -83,6 +83,16 @@ object DataType {
       metadata.toList.flatten.map(_.toCampaignMetadata(campaignId))
   }
 
+  final case class CreateRetryCampaign(name: String,
+                                       update: UpdateId,
+                                       group: GroupId,
+                                       mainCampaignId: CampaignId,
+                                      ) {
+    def mkCampaign(ns: Namespace): Campaign =
+      CreateCampaign(name, update, NonEmptyList.one(group), Some(mainCampaignId)).mkCampaign(ns)
```

Comment:
I'd simply copy over the code from `CreateCampaign::mkCampaign` method instead of re-using it because:
1. You won't need a `groupId` anymore
2. There is no complex logic in `CreateCampaign::mkCampaign` that we need to re-use

Reply:
Nevermind, just saw that you already did it.